serving/serving.go

```diff
@@ -116,9 +116,6 @@ func (serv *FeatureServer) TrainingTestSplit(stream pb.Feature_TrainingTestSplit
 				featureObserver.SetError()
 				return err
 			}
-			if isTestFinished && isTrainFinished {
```

Comment:
How do we differentiate when the entire stream should be done versus one of the iterators is done? Does this still do that? Or do we not need to do that?